torch/_functorch/aot_autograd.py

```diff
@@ -1647,6 +1587,106 @@ def remove_dupe_metadata(
         traced_tangents=traced_tangents,
     )
 
+# Given our ViewAndMutation metadata, this fn constructs a new set of metadata,
+# after adding synthetic base arguments to the function.
+# Most of the work in this fn is slogging through all of the metadata corresponding to inputs,
+# and updating it with our synthetic base calling convention.
+#
+# When config.debug_assert is set, we automatically regenerate the metadata
+# and compare it to this output for sanity.
+#
+# In addition to the updated metadata, also return the list of input indices
+# that will need to be updated in the synthetic base epilogue
+def create_synthetic_base_metadata(
+    m: ViewAndMutationMeta,
+    # Maps each outer argument idx to its inner idx (or, if this outer arg is generated from a
+    # synthetic base, you get a tuple of (i, TensorMeta), telling you the base tensor idx, and view metadata)
+    synthetic_base_info: List[Union[int, Tuple[int, torch.Tensor]]],
+    flat_args: List[Any],
+    flat_args_with_synthetic_bases: List[Any],
+) -> Tuple[ViewAndMutationMeta, List[int]]:
+
+    # For each arg in the inner calling convention, map it to its idx in the outer calling convention
+    # (there might be multiple indices, if the inner arg is a synthetic base)
+    synthetic_base_to_indices: List[List[int]] = []
+    for idx in range(len(flat_args_with_synthetic_bases)):
```

Comment:
If you like inner/outer, we should probably name this `inner_args` (and then `flat_args` is `outer_args`). Not sure if I like the name best but it is certainly more compact.

Reply:
And then, you don't say `idx`, you say `inner_idx` (?)

Reply:
yeah i don't love it either but ill stick with it for consistency